GUIDE.md

```diff
@@ -0,0 +1,1081 @@
+
+The available configuration to a Roc command is exactly what `-h` outputs. Some examples from `roc dev -h` are:
+
+```
+--build-name        The name of the generated application bundle.
+--build-path        The basepath for the application.
+--build-reducers    The reducers to use if no entry file is given, will use default entry files.
+```
+
+For clarity we will demonstrate how we can change the `--build-name` setting using the three different configuration methods.
+
+### Modify the build name directly on the CLI
+
+`roc dev --build-name="different-name"`
+
+Settings defined directly on the CLI like this will always have **highest priority**
+
+### Modify the build name with an environment variable
+
+```
+$ export ROC_CONFIG_SETTINGS='{ "build": { "name": "different-name" } }';
+$ roc dev
+```
+
+ROC_CONFIG_SETTINGS are deep-merged on top of settings from roc.config.js
+
+### Modify the build name permanently from the project using roc.config.js
+
+```js
+{
+  //... other things
+  build: {
+    name: "different-name"
+  }
+}
+```
+
+How you choose to set your configration is up to you, but in general we **highly recommend**:
+- Use CLI parameters for temporary configuration and experiments
+- `roc.config.js` for configuration that must follow your project permanently; and have it checked into source control with the rest of the project.  
+
+Only use the environmental `ROC_CONFIG_SETTINGS` in situations where you for some reason have no other choice. For example certain CI servers or sandboxed environments.  
+The Roc CLI will actually warn you when settings are appended using `ROC_CONFIG_SETTINGS` to help avoid unexpected implicit behaviour as a result of these being set. It is easy to forget about previously defined environment variables.
+
+![env warning](envwarning.png)
+
+The tooling configuration that we just introduced takes care of build-time information as well as some framework-specific settings controlled by `web-app-react`. We recommend you spend a little time to get familiar with them.
+
+You should _not_ mix your own application-specific `Runtime configuration` with this tooling configuration. They are handled separately by design.
+
+## Runtime configuration
+
+Most applications need to bring their own unique sets of configurations that can also vary depending on the environment. Examples of this can be
+- Host of backend API servers to use
+- Feature flags for the application
+- Language settings
+
+Hardcoding configuration values like this that can vary in different environments is not a great idea.
+
+To solve this in a good manner, `web-app-react` includes and exposes [node-config](https://github.com/lorenwest/node-config) to your project. It takes care of all your configuration needs.
+
+### Defining runtime configuration with node-config
+Create a new directory `config/` in your project root. Here you may define `js` or `json` files named like the corresponding environment (`NODE_ENV`) of choice. For example:
```

Comment:
You can define many other formats as well, like YAML.

Reply:
Sure. Can link to the appropriate docs.